src/streaming.ts

```diff
@@ -177,21 +177,47 @@ export class StreamProcessor {
       if (msg.domain === 'flag') {
         this.msgProcessor(
           msg,
-          this.api.getFeatureConfigByIdentifier.bind(this.api),
+          this.getFeatureConfigByIdentifierWithRetries(3),
           this.repository.setFlag.bind(this.repository),
           this.repository.deleteFlag.bind(this.repository),
         );
       } else if (msg.domain === 'target-segment') {
         this.msgProcessor(
           msg,
-          this.api.getSegmentByIdentifier.bind(this.api),
+          this.getSegmentByIdentifierWithRetries(3),
           this.repository.setSegment.bind(this.repository),
           this.repository.deleteSegment.bind(this.repository),
         );
       }
     }
   }
 
+  private getFeatureConfigByIdentifierWithRetries(maxRetries: number) {
+    for (let i = 0; i < maxRetries; i++) {
```

Comment:
It's not. I haven't added any backoff behaviour to the retry. I wasn't sure if it was necessasry for lambda users, because as soon as the instance unfreezes it'll get a timeout error immediately and the first retry will connect as normal

Reply:
I _ could_ add a backoff to each retry if desired

Reply:
i was just thinking about the non-serverless use cases - is it as simple as adding sleep() or with that block the event loop?

Reply:
Will see what the idomatic approach is for node now

Reply:
Yeah the approach is to do the following:

await new Promise(res => setTimeout(res, delay));

That won't block the event loop. It can, however, cause a resource contention issue if the function is doing loads of retries -i.e. in a lambda environment. So I'd opt to leave it out just in case. 
The function will retry max 5 times, and if it fails, we won't try anymore.